### PR TITLE
Remove reset service support from the storage backend [DPP-804]

### PR DESCRIPTION
**Pull request by mziolekda**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/indexer-benchmark/src/main/scala/ledger/indexerbenchmark/IndexerBenchmark.scala [hunk 1]
```diff
@@ -90,7 +89,7 @@ class IndexerBenchmark() {
               .migrateOnly(
                 jdbcUrl = config.indexerConfig.jdbcUrl
               )
-              .map(_ => indexerFactory.initialized())(indexerEC),
+              .map(_ => indexerFactory.initialized(loggingContext))(indexerEC),
```
Comment (rautenrieth-da) [inline ledger/indexer-benchmark/src/main/scala/ledger/indexerbenchmark/IndexerBenchmark.scala:92]: Was this change necessary? `loggingContext` was previously used as an implicit value.

**Review by rautenrieth-da — APPROVED** [review]
Nice to see all that code gone!

File: ledger/indexer-benchmark/src/main/scala/ledger/indexerbenchmark/IndexerBenchmark.scala (hunk 1 above)
Comment (mziolekda) [reply ledger/indexer-benchmark/src/main/scala/ledger/indexerbenchmark/IndexerBenchmark.scala:92]: Sadly it is necessary. Won't compile otherwise :shrug:

File: ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala [hunk 2]
```diff
@@ -20,35 +20,26 @@ import com.daml.platform.store.DbType.{
...
       readService: state.ReadService,
-      servicesExecutionContext: ExecutionContext,
       metrics: Metrics,
       lfValueTranslationCache: LfValueTranslationCache.Cache,
   )(implicit materializer: Materializer) {
 
-    def initialized(
-        resetSchema: Boolean = false
-    )(implicit loggingContext: LoggingContext): ResourceOwner[Indexer] = {
+    def initialized(implicit loggingContext: LoggingContext): ResourceOwner[Indexer] = {
```
Comment (kamil-da) [inline ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:37]:
```suggestion
    def initialized()(implicit loggingContext: LoggingContext): ResourceOwner[Indexer] = {
```

@mziolekda @rautenrieth-da This should make the change in indexer-benchmark unnecesary.

Comment (mziolekda) [reply ledger/participant-integration-api/src/main/scala/platform/indexer/JdbcIndexer.scala:37]: :+1:

**Review by SamirTalwar-DA — APPROVED** [review]
Awesome.
